goosebit/uri.py

```diff
@@ -0,0 +1,44 @@
+import copy
+import dataclasses
+import urllib.parse
+from pathlib import Path
+
+
+@dataclasses.dataclass
+class RemoteFirmwareURI:
```

Comment:
Instead of all of this I would introduce a `Firmware` DB model with at least a `name`, `version` and `uri` (can be local or remote) field. In Django you can also add methods to Models if needed, don’t know about this ORM. AFAICT the model would replace `FirmwareArtifact` and return either an URL to a local or a remote file depending on the value of the `uri` model field.

Reply:
This is a good idea, let me see what I can do.  Should be no problem adding additional functionality to the model, it is represented as a class like any other ORM.

Reply:
I think the UI should also probably change based on whether gooseBit is configured for local or remote file handling, then you can just enter a URL to the file or upload a file.